File: api/schemas/billing.py
from pydantic import BaseModel
from typing import Optional


class CheckoutSessionRequest(BaseModel):
    price_id: Optional[str] = None
    quantity: Optional[int] = None
    success_url: Optional[str] = None
    cancel_url: Optional[str] = None


class CheckoutSessionResponse(BaseModel):
    url: str


class PortalSessionResponse(BaseModel):
    url: str

File: frontend/src/api/billing.ts
import { request } from './http';

export interface CheckoutSessionRequest {
  price_id?: string;
}

export interface CheckoutSessionResponse {
  url: string;
}

export interface PortalSessionResponse {
  url: string;
}

export async function createCheckoutSession(priceId?: string): Promise<CheckoutSessionResponse> {
  const body: CheckoutSessionRequest = {};
  if (priceId) {
    body.price_id = priceId;
  }

  return request<CheckoutSessionResponse>('/billing/checkout-session', {
    method: 'POST',
    body: JSON.stringify(body),
  });
}

export async function createPortalSession(): Promise<PortalSessionResponse> {
  return request<PortalSessionResponse>('/billing/portal-session', {
    method: 'POST',
    body: JSON.stringify({}),
  });
}